### PR TITLE
Project setup

### PR DESCRIPTION
**Pull request by shreaker**
# Project setup
* cmake
----------
* clang-format
* clang-tidy
* iwyu
* valgrind
* travis CI
----------
* gitignore
----------
* catch2
* trompeloeil
----------
* Example code
	- MyGame
	- engine
	- shared

### CODE REVIEW

**Comment by shreaker** [comment]
@deep-freeze please review project setup

File: src/libengine/include/engine/Engine.hpp [hunk 1]
```diff
@@ -0,0 +1,39 @@
+#pragma once
```
Comment (shreaker) [inline src/libengine/include/engine/Engine.hpp]: @deep-freeze please help with compile errors

File: src/libengine/src/Engine.cpp [hunk 2]
```diff
@@ -0,0 +1,80 @@
+#include "engine/Engine.hpp"
```
Comment (shreaker) [inline src/libengine/src/Engine.cpp]: @deep-freeze please help with compile errors

**Comment by shreaker** [comment]
@Frozen-void @deep-freeze please review PR for merge

**Review by deep-freeze — APPROVED** [review]

**Pull request merged**